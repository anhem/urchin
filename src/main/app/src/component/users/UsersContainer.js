import React, {Component} from 'react'
import {connect} from 'react-redux'
import {getUsers} from '../../action/userAction'
import Users from "./Users";

export class UsersContainer extends Component {

    componentWillMount() {
        this.props.getUsers();
    }

    render() {
        return (
            <Users users={this.props.users}/>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getUsers: () => {
            dispatch(getUsers())
        }
    }
};

const mapStateToProps = (state) => {
    return {
        users: state.userData.users || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(UsersContainer)