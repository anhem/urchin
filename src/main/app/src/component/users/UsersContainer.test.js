import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {UsersContainer} from "./UsersContainer";

describe('UsersContainer', () => {

    let props = {
        getUsers: jest.fn()
    };

    it('match snapshot', () => {
        expect(toJson(shallow(<UsersContainer {...props}/>))).toMatchSnapshot();
    });
});
