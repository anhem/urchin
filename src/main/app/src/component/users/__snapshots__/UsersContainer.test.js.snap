// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersContainer match snapshot 1`] = `<Users />`;
